cf-agent/package_module.c

```diff
@@ -742,14 +742,17 @@ int UpdatePackagesDB(Rlist *data, const char *pm_name, UpdateType type)
         char *inventory_list = BufferClose(inventory_data);
 
         /* We can have empty list of installed software or available updates. */
-        if (!inventory_list)
+        if (!inventory_list || inventory_list[0] == '\0')
         {
             WriteDB(db_cached, inventory_key, "\n", 1);
         }
         else
         {
             WriteDB(db_cached, inventory_key, inventory_list,
                     strlen(inventory_list));
+        }
+        if (inventory_list)
```

Comment:
Might as well do an explicit comparison to `NULL` in these two modified conditions now that we are actually changing them.